shotover-proxy/tests/cassandra_int_tests/schema_awaiter.rs

```diff
@@ -5,22 +5,30 @@ use scylla::{Session, SessionBuilder};
 // But for cases like adding a new function we hit issues where the function is not yet propagated to all nodes.
 // So we make use of the scylla drivers await_schema_agreement logic to wait until all nodes are on the same schema.
 pub struct SchemaAwaiter {
-    session: Session,
+    session: Option<Session>,
 }
 
 impl SchemaAwaiter {
     pub async fn new(node: &str) -> Self {
         SchemaAwaiter {
-            session: SessionBuilder::new()
-                .known_node(node)
-                .user("cassandra", "cassandra")
-                .build()
-                .await
-                .unwrap(),
+            session: Some(
+                SessionBuilder::new()
+                    .known_node(node)
+                    .user("cassandra", "cassandra")
+                    .build()
+                    .await
+                    .unwrap(),
+            ),
         }
     }
 
+    pub fn new_noop() -> Self {
+        SchemaAwaiter { session: None }
+    }
```

Comment:
I'm just curious, why have a no-op awaiter instead of not having an awaiter (i.e. Option<SchemaAwaiter>)?
Is it so we don't end up with confusing `None`s as positional arguments? I'm not familiar with Rust idioms.

Reply:
It makes implementing test cases easier, the test case can just call await_schema_agreement without caring whether the particular test needs an await or not.

Usually putting the Option outside of the type would make more sense as its more composable.
I consider what I've done here to be an unusual case.